packages/amplify-category-api/commands/api/add-graphql-datasource.js

```diff
@@ -106,15 +107,16 @@ module.exports = {
         const apiDirPath = `${projectBackendDirPath}/${category}/${resourceName}`;
         fs.ensureDirSync(apiDirPath);
         const graphqlSchemaFilePath = `${apiDirPath}/schema.graphql`;
-        fs.ensureFileSync(graphqlSchemaFilePath);
-        const graphqlSchemaRaw = fs.readFileSync(graphqlSchemaFilePath, 'utf8');
-        const currGraphQLSchemaDoc = graphql.parse(graphqlSchemaRaw);
-
         const rdsGraphQLSchemaDoc = graphqlSchemaContext.schemaDoc;
 
+        // to check for both schema.graphql file or schema directory and dump RDS schema
+        const graphqlSchemaRaw = readSchema(apiDirPath, rdsGraphQLSchemaDoc);
+
+        const currGraphQLSchemaDoc = graphql.parse(graphqlSchemaRaw);
         const concatGraphQLSchemaDoc = mergeTypes([currGraphQLSchemaDoc, rdsGraphQLSchemaDoc], { all: true });
 
         fs.writeFileSync(graphqlSchemaFilePath, concatGraphQLSchemaDoc, 'utf8');
```

Comment:
When there is a `schema` folder, `readSchema` also ends up writing a new file called `rds.graphql`, why are we writing it once again here to `${apiDirPath}/schema.graphql`?